### PR TITLE
Implement 'use strict';

### PR DESCRIPTION
**Pull request by BobVarioa**
Fixes some of #8 

Also fixes a bug in runner which was impacting my testing of this. This doesn't hit all the things that strict mode does, but fixes the todos for it present in the code.

### CODE REVIEW

File: compiler/codegen.js [hunk 1]
```diff
@@ -1767,6 +1767,14 @@ const disposeLeftover = wasm => {
 const generateExp = (scope, decl) => {
   const expression = decl.expression;
 
+  if (expression.type === 'Literal' && typeof expression.value === 'string') {
+    if (expression.value === 'use strict') {
+      if (scope.noStrict) return internalThrow(scope, 'SyntaxError', 'Illegal "use strict" directive', false);
```
Comment (CanadaHonk) [inline compiler/codegen.js:1772]: Curious, shouldn't the parser handle this?

Comment (BobVarioa) [reply compiler/codegen.js:1772]: Some parsers have a .directive, and Babel even has some more complex things, but in general? Nothing stable.

Comment (CanadaHonk) [reply compiler/codegen.js:1772]: I meant that parsers should be able to detect this scenario and error for us, not directives generally.

Comment (BobVarioa) [reply compiler/codegen.js:1772]: Ohhh, in my testing, no. But, maybe other parsers besides acorn do?

Comment (BobVarioa) [reply compiler/codegen.js:1772]: Oh wait I misread, hmm strange.

Comment (BobVarioa) [reply compiler/codegen.js:1772]: After some more testing, I'm coming to the same result. I'll go ahead and remove it.

**Pull request merged**